### PR TITLE
Make income details update notification consistent across user account

### LINKED ISSUE #2056: Make `INCOME_DETAILS_UPDATE` notification consistent across user account
### Problem

The `INCOME_DETAILS_UPDATE` notification on the dashboard seems to be per-device / per-browser. 

1. Dismissed notification & updated income details in browser1. 
2. Opened browser2 and signed in. 
3. Notification visible at top of dashboard on browser2, even after hard refresh. Notification remains dismissed (i.e. not visible) on browser1.
updateIncomeDetails

### Solution

Ensure that the `INCOME_DETAILS_UPDATE` notification on the dashboard is consistent across a user's account (rather than per-device / per-browser).

### PR DESCRIPTION
**Pull request by Silver-IT**
### Summary of Changes

- Closes #2056

### CODE REVIEW

**Review by Silver-IT — COMMENTED** [review]
Ready for the review!

File: frontend/controller/actions/identity.js [hunk 1]
```diff
@@ -460,20 +459,21 @@ export default (sbp('sbp/selectors/register', {
...
-            .find(cID => has(state[identityContractID].groups, cID))
+          const gId = Object.keys(state.contracts).find(cID => has(state[identityContractID].groups, cID))
 
           if (gId) {
             sbp('gi.actions/group/switch', gId)
           }
         }
+
+        // NOTE: load users preferences config which is saved in KV store
+        await sbp('gi.actions/kv/loadPreferences')
```
Comment (taoeffect) [inline frontend/controller/actions/identity.js]: Would it be OK to move this up next to the other `kv/load` statement on line 398?

File: frontend/controller/actions/kv-stores.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
+'use strict'
```
Comment (taoeffect) [inline frontend/controller/actions/kv-stores.js]: Nice job splitting these out into a new file! I agree they belong in a separate file.

Could you please make it even more organized by creating separate scoped files? 

For example, for each KV store that you create, a separate file:

E.g. `group-kv.js` (all group contract related KV selectors go here), and `identity-kv.js` (all identity contract related selectors go here).

Also, I think we should include the name of the type of contract in the selector itself to make things even more clear, so e.g.:

```
'gi.actions/kv/fetchChatRoomUnreadMessages'
```

Would get renamed to:

```
'gi.actions/identity/kv/fetchChatRoomUnreadMessages'
```

And:

```
'gi.actions/kv/updateLastLoggedIn'
```

Would get renamed to:

```
'gi.actions/group/kv/updateLastLoggedIn'
```

This makes it super clear which KV store is being dealt with.

File: frontend/controller/actions/identity.js [hunk 3]
```diff
@@ -803,140 +803,15 @@ export default (sbp('sbp/selectors/register', {
...
-  },
+  // NOTE: Adding the following 4 actions is a temporary solution and needs to be removed
+  //       when no older (than 0.5.1) version of contracts are being used
+  //       https://okturtles.slack.com/archives/C0EH7P20Y/p1718990127660609
+  ...Object.fromEntries([
+    'addChatRoomUnreadMessage',
+    'initChatRoomUnreadMessages',
+    'removeChatRoomUnreadMessage',
+    'deleteChatRoomUnreadMessages'
+  ].map(fnName => [`gi.actions/identity/${fnName}`, (...args) => sbp(`gi.actions/kv/${fnName}`, args)])),
```
Comment (taoeffect) [inline frontend/controller/actions/identity.js]: I appreciate that you did this. The thing is, it turns out that the problems on the server were related to real bugs in the internals/core of Chelonia, and we're gonna have to recreate the group anyway, so now is a good time to organize these selectors.

Please feel free to delete this section and refactor the selectors as needed. Also, please see my comment below about organizing into `identity-kv.js` etc. type files.

**Review by taoeffect — CHANGES_REQUESTED** [review]
Great work @Silver-IT! This is quite a PR!

I left a few change requests.

Also I tested this with mobile (using `grunt dev --tunnel`) and it seemed to work correctly (when I dismissed the banner on desktop, it also dismissed it on the phone).

However, when I ran this command on desktop to try and get the banner to re-show, it didn't work:

```js
sbp('gi.actions/kv/updateDistributionBannerVisibility', { contractID: sbp('state/vuex/state').currentGroupId, hidden: false })
```

Why is that?

Also, could you please merge `master` into this PR so that this error goes away?

Screenshot 2024-06-25 at 11 04 06 AM

File: frontend/controller/actions/identity.js (hunk 1 above)
Comment (Silver-IT) [reply frontend/controller/actions/identity.js]: Yeah, it's fine.

File: frontend/controller/actions/kv-stores.js (hunk 2 above)
Comment (Silver-IT) [reply frontend/controller/actions/kv-stores.js]: Makes sense. Nice suggestion!

**Comment by Silver-IT** [comment]
> However, when I ran this command on desktop to try and get the banner to re-show, it didn't work:
> ```
> sbp('gi.actions/kv/updateDistributionBannerVisibility', { contractID: sbp('state/vuex/state').currentGroupId, hidden: false })
> ```
> Why is that?

I can't understand why it doesn't work. Actually, it works well on my side.

File: frontend/model/state.js [hunk 4]
```diff
@@ -28,7 +28,8 @@ const initialState = {
   namespaceLookups: Object.create(null), // { [username]: sbp('namespace/lookup') }
   periodicNotificationAlreadyFiredMap: {}, // { notificationKey: boolean },
   contractSiginingKeys: Object.create(null),
-  lastLoggedIn: {} // Group last logged in information
+  lastLoggedIn: {}, // Group last logged in information
+  preferences: {}
```
Comment (taoeffect) [inline frontend/model/state.js:32]: Any time we modify the state like this we need to add an entry in `postUpgradeVerification`

**Review by taoeffect — APPROVED** [review]
I hope it's OK with you that I added the `postUpgradeVerification` fix to get this merged! Great job @Silver-IT !

**Pull request merged**